### PR TITLE
Enables shuffling in iterative_train_test_split

### PR DESCRIPTION
**Pull request by northern-64bit**
Moving the PR (https://github.com/scikit-multilearn/scikit-multilearn/pull/290) made by @erikhuck in scikit-multilearn.

### CODE REVIEW

**Comment by northern-64bit** [comment]
Hello @mbbyn! Can you please specify in greater detail what you ment with the comment (here https://github.com/scikit-multilearn/scikit-multilearn/pull/290#issuecomment-1872280469)? I added the following change in order to address it: https://github.com/scikit-multilearn-ng/scikit-multilearn-ng/pull/8/files#diff-ce9a65da687b67daef369761039bc521468402ca622ba166f038bd699bbb76a5R104

**Comment by mbbyn** [comment]
Hello @northern-64bit 👋 

If you notice, the `iterative_train_test_split` function simply instantiates the `IterativeStratification` class under the hood and passes it the `shuffle` and `random_state` arguments.

The PR in question is fixing the lack of shuffling in the function, but a better fix IMO would be to update the class instead. This way, people using the `IterativeStratification` class would also get the shuffle fix (which doesn't exist in the class yet).

For example, please see how the main scikit-learn library uses the `shuffle` in the class itself (although I think they should also use `random_state` as well for reproducible results).

https://github.com/scikit-learn/scikit-learn/blob/678e3999eeaadcaaef523d1d0d2f52a25986d460/sklearn/model_selection/_split.py#L764-L765

**Comment by northern-64bit** [comment]
Hello @mbbyn!

Thanks for the more detailed explanation! I have now moved the changes to `IterativeStratification` in `_iter_test_indices`, which is accessed every time .split() is used. So it should now work for all uses of `IterativeStratification`.
